support-services/src/main/scala/com/gu/aws/AwsCloudWatchMetricPut.scala

```diff
@@ -63,11 +63,11 @@ object AwsCloudWatchMetricPut {
         MetricDimensionName("Environment") -> MetricDimensionValue(environment.toString)
       ))
 
-  def paymentSuccessRequest(stage: Stage, paymentProvider: PaymentProvider, productType: ProductType): MetricRequest =
+  def paymentSuccessRequest(stage: Stage, paymentProvider: Option[PaymentProvider], productType: ProductType): MetricRequest =
```

Comment:
doesn't really make sense for a paymentSuccessRequest to happen when there is no payment provider.  So it should only be called if there is a payment provider.

Reply:
Are you sure about this? It is called by the SendAcquisitionEvent lambda which suggests that it relates to a successful sign up rather than a payment event but I don't really know what it's used for so it's unclear whether we want to register anything here or not

Reply:
no I realised later and made another comment and forgot to come back and edit this one

> as per earlier, looks like it does make sense for no payment method, but it should be more like subscriptionSuccessRequest. Ideally the metric should be renamed also, but there would be a bit more work checking the alarms as well.

so it's more like a rename needed to make it reflect what it actually does.  I know refactoring inherently has a risk, but it also has benefits for future clarity.

Reply:
I'm going to leave it out of this PR as the knock on effects could be quite extensive

Reply:
good thinking better as a separate PR 👍